microindex/ztests/manual.yaml

```diff
@@ -1,14 +1,14 @@
 script: |
-  microindex create -o index -k s babble.tzng
-  zq -f tzng index.zng > index.tzng
-  zq -o sorted.zng "count() by s | put key=s | cut key | sort key" babble.tzng
-  microindex convert -o manual -k key sorted.zng
-  zq -f tzng manual.zng > manual.tzng
-  diff index.tzng manual.tzng
+  microindex create -o index.zng -k s babble.zson
+  zq -z index.zng > index.zson
+  zq -o sorted.zng "count() by s | put key=s | cut key | sort key" babble.zson
+  microindex convert -o manual.zng -k key sorted.zng
```

Comment:
Here's the material change. The lack of the `.zng` extensions on the `-o` options to `microindex create` was making for zero-length `index.tzng` and `sorted.zng` files. So those zero-length files `diff`'ed successfully and made it look like the test was doing something real and passing when in fact it was effectively a no-op.

I can't help but see this base `zq` behavior as buggy. Given how we want everything to be stream-friendly, I guess I could respect it as expected behavior to output a zero-length stream if a zero-length stream is given as input. OTOH, if an explicit filename is given as the source of that input stream and that file doesn't even exist, it seems like we should give some kind of error/warning or non-zero exit code. Thoughts?

```
$ ls -l doesnotexist
ls: doesnotexist: No such file or directory

$ zq -z doesnotexist > something

$ ls -l something
-rw-r--r--  1 phil  staff  0 Mar 19 09:03 something

$ echo $?
0
```

Reply:
Yes, this is a bug and appears to be a vestige about how the command parsing logic evolved.  What happens here is "doesnotexist" is compiled as a Z query and the rest of the code is perfectly happy with the fact that array of input files is length 0 so nothing happens.  We just need to a check for no files (stdin appears as "-" and doesn't cause this problem obviously).

Reply:
I could swear we used to have a check for this...

Reply:
Fixed in #2382